myhoard/controller.py

```diff
@@ -1196,6 +1216,45 @@ def _state_file_from_stream_id(self, stream_id):
         safe_stream_id = stream_id.replace(":", "_").replace(".", "_")
         return os.path.join(self.state_dir, f"{safe_stream_id}.json")
 
+    def _switch_basebackup_if_possible(self):
+        # We're trying to restore a backup but that keeps on failing in the basebackup restoration phase.
+        # If we have an older backup available try restoring that and play back all binlogs so that the
+        # system should end up in the exact same state eventually.
+        backups = sorted(
+            (backup for backup in self.state["backups"] if backup["completed_at"]), key=lambda d: d["completed_at"]
+        )
+        current_stream_id = self.state["restore_options"]["stream_id"]
+        earlier_backup = None
+        for backup in backups:
+            if backup["stream_id"] == current_stream_id:
+                break
+            else:
+                earlier_backup = backup
+        else:
+            raise Exception(f"Stream {current_stream_id} being restored not found in completed backups: {backups}")
```

Comment:
It could be beneficial to raise something specific here so that if there is external python code calling myhoard directly, a specific exception could be expected for this case, instead of falling back to matching error message string.

Reply:
Ah true; I was expecting this to propagate further up.